#161996678 Add a user sign up and database storage.

BrianSerem opened the pull request:
#### What does this PR do?
Add a user sign up through storage of info in an external database.

#### Description of Task to be completed?
Have the following endpoint working with storage of data in a database.
/api/v1/auth/signup

#### How should this be manually tested?
Clone this repository.
Cd into the directory.
Export run.py as a flask application
flask run and use the given url to test the endpoint.

#### Any background context you want to provide?
Every storage of data is now on a database.
After a successful sign up, the user is authenticated and given a unique access token for accessing restricted areas of the app.
 
#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/161996678

#### Screenshots
![userregistration](https://user-images.githubusercontent.com/44457835/48817211-69097d00-ed56-11e8-980d-f01e74782e7f.png)

BrianSerem committed:
[Feature #161996678] Added a user sign up and database storage.

BrianSerem committed:
[Feature #161996678] Added a user sign up and database storage.

BrianSerem committed:
--amend

cosmas28 commented on app/__init__.py:
```diff
@@ -3,14 +3,17 @@
 def create_app():
     """Crete app fuction."""
     app = Flask(__name__)
+    app.secret_key = "brian"
```
This secret key should be moved to dotenv file. This is a good practice.

BrianSerem replied:
I created a variable secret_key inside the .env file

cosmas28 commented on app/api/V2/models/usermodels.py:
```diff
@@ -1,36 +1,57 @@
 """COntain User model."""
```
kindly fix the typos.

BrianSerem replied:
Fixed the typo in the usermodels.py  header doctsring

cosmas28 commented on app/api/V2/__init__.py:
```diff
@@ -7,7 +7,7 @@
+from .views.userviews import UserRegistration
 
 version_2 = Blueprint('v2', __name__)
 
```
there should be only one line space at the end of the files. Let this be the last time I'm repeating this feedback.

BrianSerem replied:
Deleted the extra lines at the end of the file

cosmas28 commented on app/api/V2/models/usermodels.py:
```diff
@@ -1,36 +1,57 @@
+        return user
+
+
+   
```
Too many lines at the end of this file. Kindly leave one line only.

BrianSerem replied:
removed the extra pieces of lines at the end of the file

cosmas28 commented on app/api/V2/models/usermodels.py:
```diff
@@ -1,36 +1,57 @@
-        for user in users:
-            if user.username == username:
-                return user
+        cur = connection.cursor()
```
Kindly use a more descriptive variable name than `cur`.

BrianSerem replied:
I have used cursorobject for the previously "cur" variable

cosmas28 commented on app/api/V2/models/usermodels.py:
```diff
@@ -1,36 +1,57 @@
 
-    def get_by_username(self, username):
+    @classmethod
+    def get_by_username(cls, username):
```
This function is not descriptive enough. I think it should be `get_user_by_username`.

BrianSerem replied:
I changed its name to get_user_by_username as you suggested

cosmas28 commented on app/api/V2/views/userviews.py:
```diff
@@ -0,0 +1,35 @@
+            return {"access token":access_token}, 200
+            return {"error":"error connecting to the database"}, 404
+
+        return{"error":"username {} is already taken".format(data['username'])}
```
Leave a line space at the end of each file.

BrianSerem replied:
Left a single a the end of all files in this PR

cosmas28 commented on app/api/V2/views/userviews.py:
```diff
@@ -0,0 +1,35 @@
+            access_token = create_access_token(identity=user.id, fresh=True)
+            refresh_token = create_refresh_token(user.id)
+            return {"access token":access_token}, 200
+            return {"error":"error connecting to the database"}, 404
```
You can only have one return statement in a block of code like `if`. The second return statement can't work.

BrianSerem replied:
Removed the extra return statement and placed it inside a try block.

BrianSerem committed:
chore(user sign up): worked on feedback
-added one line to files misisng at the end
-removed extra lines in some files.
-Added more requirements to requirements file
[Delivers ##162113373i]
